app/controllers/api/custom_button_sets_controller.rb

```diff
@@ -7,5 +7,18 @@ def create_resource(type, id, data)
     def edit_resource(type, id, data)
       super(type, id, data.deep_symbolize_keys)
     end
+
+    def options
+      render_options(:custom_button_sets, build_custom_button_sets_options)
+    end
+
+    def build_custom_button_sets_options
+      {
+        :custom_button_sets => {
+          :generic_object_definitions => CustomButtonSet.find_all_by_class_name('GenericObjectDefinition'),
```

Comment:
If this returns `CustomButtonSet` objects - I'm wondering if this is appropriate for an OPTIONS request, which should only return metadata. Perhaps this could be solved with a filter on the GET #index call?

Reply:
Yes, I suggested something on similar lines above -

```
GET /api/custom_button_sets?expand=resources&attributes=set_data
```

If it were possible to add a filter as shown below, that would address my requirement -
```
filter[]=set_data[:applies_to_class]=GenericObject
```